src/main/scala/mesosphere/marathon/api/CORSFilter.scala

```diff
@@ -0,0 +1,46 @@
+package mesosphere.marathon.api
+
+import javax.inject.{ Inject }
+import javax.servlet._
+import javax.servlet.http.{ HttpServletRequest, HttpServletResponse }
+import mesosphere.marathon.{ MarathonConf }
+
+import scala.collection.JavaConverters._
+
+class CORSFilter @Inject() (config: MarathonConf) extends Filter {
+  // Map access_control_allow_origin flag into separate headers
+  lazy val origins: Seq[String] = config.accessControlAllowOrigin()
```

Comment:
Consider this change:

```scala
lazy val origins: Option[Seq[String]] =
  config.accessControlAllowOrigin.get.map { configValue =>
    configValue.split(",").map(_.trim)
  }
```

Then your if test in the match in `doFilter` can simplify to `origins.isDefined`!